composer/trainer/trainer.py

```diff
@@ -1204,35 +1226,73 @@ def _get_autoresume_checkpoint(
         Returns:
             Optional[str]: The path to the latest checkpoint, if found, otherwise None.
         """
+        log.debug(
+            f'Entering _get_autoresume_checkpoint with save_latest_remote_file_name: {save_latest_remote_file_name}')
         save_latest_filename = format_name_with_dist(save_latest_filename, self.state.run_name)
         save_folder = format_name_with_dist(save_folder, self.state.run_name)
         save_latest_remote_file_name = format_name_with_dist(save_latest_remote_file_name, self.state.run_name)
         latest_checkpoint_path = os.path.join(save_folder, save_latest_filename)
 
+        log.info(
+            f'Looking for autoresume checkpoint: {save_latest_remote_file_name} (remote), {latest_checkpoint_path} (local)'
+        )
+
         # If latest checkpoint is not saved locally, try to fetch from loggers
-        if not os.path.exists(latest_checkpoint_path):
-            # Make save folder in case it doesn't exist so latest checkpoint can be downloaded
+        if not os.path.exists(latest_checkpoint_path) and (dist.get_global_rank() == 0 or self.deepspeed_enabled):
+            log.debug(f'Attempting to download the checkpoint on to rank {dist.get_global_rank()}')
```

Comment:
remove since we show a progress bar during download?

Reply:
I will leave this one because the progress bar doesn't show if the download can't be started for some reason,